colossus/src/main/scala/colossus/protocols/http/Header.scala

```diff
@@ -23,14 +23,18 @@ object HttpMethod {
 
   val methods: List[HttpMethod] = List(Get, Post, Put, Delete, Head, Options, Trace, Connect, Patch)
 
-  def apply(str: String): HttpMethod = methods.find{_.name == str.toUpperCase}.getOrElse(
-    throw new ParseException(s"Invalid http method $str")
-  )
+  def apply(str: String): HttpMethod = {
```

Comment:
The list is so small it probably doesn't matter, but this is a pretty hot path, did you try this look up using a Map or just good ol' if statements :)